tests/pyccel/test_pyccel.py

```diff
@@ -66,6 +66,29 @@ def compile_pyccel(path_dir,test_file, options = ""):
     p.wait()
     assert(p.returncode==0)
 
+#------------------------------------------------------------------------------
+def compile_c(path_dir,test_file,dependencies):
+    root = insert_pyccel_folder(test_file)[:-3]
+
+    assert(os.path.isfile(root+".c"))
+
+    command = [shutil.which("gcc"), "-O3", "%s.c" % root]
+    if isinstance(dependencies, list):
+        for d in dependencies:
+            d = insert_pyccel_folder(d)
+            command.append(d[:-3]+".o")
+            command.append("-I"+os.path.dirname(d))
+    elif isinstance(dependencies, str):
+        dependencies = insert_pyccel_folder(dependencies)
+        command.append(dependencies[:-3]+".o")
+        command.append("-I"+os.path.dirname(dependencies))
```

Comment:
There is some repeated code here. We could write:
```python
deps = dependencies if hasattr(dependencies, '__iter__') else [dependencies]

for d in deps:
    d = insert_pyccel_folder(d)
    command.append(d[:-3]+".o")
    command.append("-I"+os.path.dirname(d))
```

Reply:
A string has a `__iter__` method. I have used:
```
deps = [dependencies] if isinstance(dependencies, str) else dependencies
```
That way the code works for tuples too